src/MichiTokenizeRequestor.sol

```diff
@@ -0,0 +1,121 @@
+        address requestor;
+        address michiWalletAddress;
+        uint256 requestId;
+    }
+
+    /// @notice address of the MichiTBALocker
+    address public michiTBALocker;
+
+    /// @notice instance of the receipt NFT
+    IMichiWalletReceiptNFT public michiWalletReceiptNFT;
+
+    /// @notice enter id to retrieve request
+    mapping(uint256 => Request) public idToRequest;
+
+    /// @notice mapping of approved Michi Wallet collections
+    mapping(address => bool) public approvedCollections;
+
+    /// @notice error emitted when an unapproved nft collection is transferred
+    error UnapprovedCollection(address collection);
+
+    /// @notice error returned when 6551 wallet owner is not sender
+    error UnauthorizedCaller(address caller);
+
+    /// @notice error returned when approving an already approved collection
+    error CollectionAlreadyApproved(address collection);
+
+    /// @notice error returned when removing an unapproved collection
+    error CollectionNotApproved(address collection);
+
+    /// @notice event emitted when a new tokenize request is created
+    event NewTokenizeRequest(address indexed requester, address indexed michiWallet, uint256 indexed requestId);
+
+    /// @notice event emitted when a new collection is added
+    event CollectionApproved(address indexed collection);
+
+    /// @notice event emitted when a collection has been removed
+    event CollectionRemoved(address indexed collection);
+
+    /// @dev Constructor for MichiTokenizeRequestor contract
+    /// @param michiTBALocker_ address of the MichiTBALocker
+    /// @param michiWalletReceiptNFT_ address of the MichiWalletReceiptNFT
+    constructor(address michiTBALocker_, address michiWalletReceiptNFT_) {
+        michiTBALocker = michiTBALocker_;
+        michiWalletReceiptNFT = IMichiWalletReceiptNFT(michiWalletReceiptNFT_);
+    }
+
+    /// @dev Initiate tokenized points request
+    /// @param michiWalletAddress tba address that holds points to be tokenized
+    function createTokenizePointsRequest(address michiWalletAddress) external {
+        // get tba ownership nft
+        (, address tokenContract, uint256 tokenId) = AccountV3Upgradable(payable(michiWalletAddress)).token();
+
+        if (!approvedCollections[tokenContract]) revert UnapprovedCollection(tokenContract);
+
+        // verify nft owner is caller
+        if (IERC721(tokenContract).ownerOf(tokenId) != msg.sender) revert UnauthorizedCaller(msg.sender);
+
+        // transfer in michi wallet nft
+        IERC721(tokenContract).safeTransferFrom(msg.sender, michiTBALocker, tokenId);
+
+        // mint the user a receipt NFT of same index
+        // in case other projects (such as AVSs) airdrop tokens to YT holders
+        michiWalletReceiptNFT.mint(msg.sender, tokenId);
+
+        _requestIds.increment();
+        uint256 newRequestId = _requestIds.current();
+        idToRequest[newRequestId] = Request(msg.sender, michiWalletAddress, newRequestId);
```

Comment:
Probably best to mint after all the checks. And validate the idToRequest[newRequestId] is indeed uninitialized.

Reply:
Moved the mint to after idToRequest is created. Is it necessary to check that it's uninitialized before updating its state as we're incrementing _requestIds beforehand?